controllers/reconcilers/mm_inferenceservice_reconciler.go

```diff
@@ -67,7 +68,11 @@ func (r *ModelMeshInferenceServiceReconciler) DeleteModelMeshResourcesIfNoMMIsvc
 
 	for i := len(inferenceServiceList.Items) - 1; i >= 0; i-- {
 		inferenceService := inferenceServiceList.Items[i]
-		if !utils.IsDeploymentModeForIsvcModelMesh(&inferenceService) {
+		isvcDeploymentMode, err := utils.GetDeploymentModeForIsvc(ctx, r.client, &inferenceService)
+		if err != nil {
+			return err
+		}
+		if isvcDeploymentMode == utils.ModelMesh {
```

Comment:
Also here, the logic looks inverted.
```suggestion
		if isvcDeploymentMode != utils.ModelMesh {
```